keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,731 @@
+
+For each device, if required, the DRA Driver processing the device allocation
+can report the status of it in the `Status.Devices` of the ResourceClaim
+by using the Kubernetes API.
+
+```golang
+// ResourceClaimStatus tracks whether the resource has been allocated and what
+// the result of that was.
+type ResourceClaimStatus struct {
+    ...
+
+    // Devices contains the status of each device allocated for this
+    // claim, as reported by the driver. This can include driver-specific
+    // information. Entries are owned by their respective drivers.
+    //
+    // +optional
+    // +listType=map
+    // +listMapKey=driver
+    // +listMapKey=device
+    // +listMapKey=pool
+    // +featureGate=DRAResourceClaimDeviceStatus
+    Devices []AllocatedDeviceStatus `json:"devices,omitempty" protobuf:"bytes,4,opt,name=devices"`
+}
+
+// AllocatedDeviceStatus contains the status of an allocated device, if the
+// driver chooses to report it. This may include driver-specific information.
+type AllocatedDeviceStatus struct {
+    // Driver specifies the name of the DRA driver whose kubelet
+    // plugin should be invoked to process the allocation once the claim is
+    // needed on a node.
+    //
+    // Must be a DNS subdomain and should end with a DNS domain owned by the
+    // vendor of the driver.
+    //
+    // +required
+    Driver string `json:"driver" protobuf:"bytes,1,rep,name=driver"`
+
+    // This name together with the driver name and the device name field
+    // identify which device was allocated (`<driver name>/<pool name>/<device name>`).
+    //
+    // Must not be longer than 253 characters and may contain one or more
+    // DNS sub-domains separated by slashes.
+    //
+    // +required
+    Pool string `json:"pool" protobuf:"bytes,2,rep,name=pool"`
+
+    // Device references one device instance via its name in the driver's
+    // resource pool. It must be a DNS label.
+    //
+    // +required
+    Device string `json:"device" protobuf:"bytes,3,rep,name=device"`
+
+    // Conditions contains the latest observation of the device's state.
+    // If the device has been configured according to the class and claim
+    // config references, the `Ready` condition should be True.
+    //
+    // +optional
+    // +listType=atomic
+    Conditions []metav1.Condition `json:"conditions" protobuf:"bytes,4,opt,name=conditions"`
+
+    // Data contains arbitrary driver-specific data.
+    //
+    // +optional
+    Data runtime.RawExtension `json:"data,omitempty" protobuf:"bytes,5,opt,name=data"`
+
+    // NetworkData contains network-related information specific to the device.
+    //
+    // +optional
+    NetworkData NetworkDeviceData `json:"networkData,omitempty" protobuf:"bytes,6,opt,name=networkData"`
+}
+
+// NetworkDeviceData provides network-related details for the allocated device.
+// This information may be filled by drivers or other components to configure
+// or identify the device within a network context.
```

Comment:
Should probably include the interface index? Name, ips, and hwaddr are all mutable,.

(In which case _maybe_ you don't need to include the interface name?)

Reply:
If you have enough privilege to rename an interface, none of the rest really matters, right?  That doesn't stop us from reporting podIPs

Reply:
I wasn't worrying just about privilege/attacks. Maybe there are use cases where the interface eventually gets renamed for some reason?

It just seems like, if it's possible to refer to the device via an immutable identifier rather than a mutable one, then we should?

Reply:
Agree with that principle

Reply:
The reference of the interface in the kubernetes cluster is: Driver + "/" + Pool + "/" + Device
The index is the reference of the interface in a Linux operating system, Devices are supposed to abstract any network interface, most commonly the linux ones, but not only those.
The drivers owns the interface, if it is renamed, then I expect the driver to update the status, but I don't expect the consumer of the api to use the index of Linux interface, so interface name sounds much better to me